### PR TITLE
Customise and cleanup create test

### PR DESCRIPTION
**Pull request by prietyc123**
**What type of PR is this?**

/kind cleanup

**What does does this PR do / why we need it**:

Just customise and remove duplicate tests.

**Which issue(s) this PR fixes**:

Fixes - Part of #3385 

**PR acceptance criteria**:

- [ ] Unit test 

- [x] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer**:

`make test-cmd-devfile-create` should run successfully.

### CODE REVIEW

File: tests/integration/devfile/cmd_devfile_create_test.go [hunk 1]
```diff
@@ -258,6 +240,14 @@ var _ = Describe("odo devfile create command tests", func() {
 		})
 	})
 
+	Context("When executing odo create with component with no devBuild command", func() {
+		It("should successfully create the devfile component", func() {
+			// Quarkus devfile has no devBuild command
+			output := helper.CmdShouldPass("odo", "create", "java-quarkus")
+			helper.MatchAllInOutput(output, []string{"Please use `odo push` command to create the component with source deployed"})
```
Comment (mik-dass) [inline tests/integration/devfile/cmd_devfile_create_test.go:247]: we should also check that run action string is present and build action string is not present.

Comment (prietyc123) [reply tests/integration/devfile/cmd_devfile_create_test.go:254]: > we should also check that run action string is present and build action string is not present.

But this test is only for `odo create` verification and I don't find anything related to run and build in `odo create java-quarkus`. I think this should be verified in `test-cmd-devfile-push` as it is purely related to that. Logs for `odo create java-quarkus`

```
$ odo preference view
PARAMETER             CURRENT_VALUE
UpdateNotification    false
NamePrefix            
Timeout               
PushTimeout           
Experimental          true
PushTarget            

$ odo create java-quarkus
Experimental mode is enabled, use at your own risk

Validation
 ✓  Checking devfile existence [87285ns]
 ✓  Checking devfile compatibility [160078ns]
 ✓  Creating a devfile component from registry: RegistryName [165146ns]
 ✓  Validating devfile component [575155ns]

Please use `odo push` command to create the component with source deployed
```

Comment (mik-dass) [reply tests/integration/devfile/cmd_devfile_create_test.go:254]: OK. I thought it was executing push too.

>  I think this should be verified in test-cmd-devfile-push as it is purely related to that. Logs for odo create java-quarkus

Do we really need an extra create test for `no build components`? I mean the test in `test-cmd-devfile-push` would execute the same thing too.

File: tests/integration/devfile/cmd_devfile_create_test.go [hunk 2]
```diff
@@ -115,8 +115,7 @@ var _ = Describe("odo devfile create command tests", func() {
 
 	Context("When executing odo create with devfile component type argument and --project flag", func() {
 		It("should successfully create the devfile component", func() {
-			componentNamespace := helper.RandString(6)
-			helper.CmdShouldPass("odo", "create", "java-openliberty", "--project", componentNamespace)
+			helper.CmdShouldPass("odo", "create", "java-openliberty", "--project", cmpName)
```
Comment (dharmit) [inline tests/integration/devfile/cmd_devfile_create_test.go:118]: We're passing component name to `--project` flag. Is that intentional?

Comment (prietyc123) [reply tests/integration/devfile/cmd_devfile_create_test.go:118]: Yes.

Comment (dharmit) [reply tests/integration/devfile/cmd_devfile_create_test.go:118]: Ack, thanks. :+1:

**Comment by prietyc123** [comment]
Yes, I have tested it locally 

/hold cancel

**Comment by amitkrout** [comment]
> Yes, I have tested it locally
> 
> /hold cancel

/lgtm

**Pull request merged**